code/components/extra-natives-five/src/InterfaceExtraNatives.cpp

```diff
@@ -21,12 +27,115 @@ struct MapZoomData
 	atArray<MapDataZoomLevel> zoomLevels;
 };
 
-static uint64_t* expandedRadar;
-static uint64_t* revealFullMap;
+static bool* expandedRadar;
+static bool* revealFullMap;
 
 static MapZoomData* zoomData;
 static std::vector<MapDataZoomLevel> defaultZoomLevels;
 
+struct MinimapData
+{
+	char name[100];
+	float posX;
+	float posY;
+	float sizeX;
+	float sizeY;
+	char alignX;
+	char alignY;
+
+private:
+	char pad[2];
+};
+
+static MinimapData* minimapArray;
+static constexpr int minimapEntries = 11;
+
+static MinimapData minimapOldEntries[minimapEntries];
+
+struct HudComponentData // see "frontend.xml"
+{
+	char unk0[8];
+
+	int index;
+	int depth;
+	int listId;
+	int listPriority;
+
+	char name[40];
+	char alignX[2];
+	char alignY[2];
+
+	float posX;
+	float posY;
+	float sizeX;
+	float sizeY;
+
+	float unk1[2];
+	float unk2[2];
+	float scriptPosX;
+	float scriptPosY;
+	float unk4[2];
+
+	uint32_t color;
+};
+
+static std::map<int, HudComponentData> hudComponentOldEntries{};
+
+static HudComponentData* hudComponentsArray;
+static uint32_t hudComponentsCount;
+
+static void EnsureHudComponentBackup(int index)
+{
+	if (index < hudComponentsCount)
+	{
+		auto& component = hudComponentsArray[index];
+
+		if (auto& it = hudComponentOldEntries.find(index); it == hudComponentOldEntries.end())
+		{
+			HudComponentData origData = component;
```

Comment:
Is there a reason for the second copy here, outside of `.insert({})` potentially being annoyed?

(if it's just the latter, would `.emplace(index, component)` work?)